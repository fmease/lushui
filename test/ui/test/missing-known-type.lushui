;;; TEST fail run
;;; TEST compiler-args --no-core -Z internals
;;; @Bug unfortunately, this is a runtime error right now

@intrinsic
data Type: Type

@intrinsic
data Nat: Type

@intrinsic
equal: Nat -> Nat -> Bool

data Bool: Type of
    false: Bool
    true: Bool

main: Bool = equal 0 1
