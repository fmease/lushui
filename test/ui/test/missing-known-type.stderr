error[E062]: the known binding ‘Bool.false’ is not defined

error: aborting due to previous error

